R/meta_regression_plots.R

```diff
@@ -0,0 +1,487 @@
+  relative_effects <- c(relative_effects, c(0.5, 8, 2))
+  contributions <- c(contributions, c(2, 3, 1))
+  
+  treatments <- c(treatments, rep("the_Dung_named", 2))
+  covariate_values <- c(covariate_values, c(1, 8))
+  relative_effects <- c(relative_effects, c(2, 4.8))
+  contributions <- c(contributions, c(2, 5))
+  
+  treatments <- c(treatments, rep("the_Great", 3))
+  covariate_values <- c(covariate_values, c(1.5, 2.5, 3.5))
+  relative_effects <- c(relative_effects, c(1.5, 2.5, 3.5))
+  contributions <- c(contributions, c(2, 3, 1))
+  
+  # the_Little is the reference treatment
+  
+  treatments <- c(treatments, rep("the_Slit_nosed", 3))
+  covariate_values <- c(covariate_values, c(2, 4, 6))
+  relative_effects <- c(relative_effects, c(6, 2, 4))
+  contributions <- c(contributions, c(2, 3, 5))
+  
+  treatments <- c(treatments, rep("the_Younger", 3))
+  covariate_values <- c(covariate_values, c(5, 7.7, 9))
+  relative_effects <- c(relative_effects, c(9, 3, 6))
+  contributions <- c(contributions, c(6, 6, 6))
+  
+  contribution_df <- data.frame(
+    Treatment = treatments,
+    covariate_value = covariate_values,
+    relative_effect = relative_effects,
+    contribution = contributions
+  )
+  
+  return(contribution_df[contribution_df$Treatment %in% comparator, ])
+}
+
+#' Find the confidence regions for the regression analysis.
+#'
+#' @param model GEMTC model result object.
+#' @param reference Name of reference treatment.
+#' @param comparator Name of comparison treatment for which to find the contributions.
+#' @param divisions The number of divisions the confidence region should be drawn in. Defaults to 10.
+#'
+#' @return Data frame containing contribution details. Each row represents a confidence interval at a specific covariate value, for a given treatment. Columns are:
+#' - Treatment: The treatment for which this confidence interval relates.
+#' - covariate_value: Value of the covariate for this interval
+#' - y_min Relative effect of the lower end of this interval
+#' - y_max: Relative effect of the upper end of this interval
+.FindRegressionConfidenceRegion <- function(model, reference, comparator, divisions = 10) {
+  
+  treatments <- c()
+  covariate_values <- c()
+  y_mins <- c()
+  y_maxs <- c()
+  
+  for (treatment in c("the_Butcher", "the_Dung_named", "the_Great", "the_Slit_nosed", "the_Younger")) {
+    start_x <- .FindRegressionStartX(model, reference, treatment)
+    end_x <- .FindRegressionEndX(model, reference, treatment)
+    intersect <- .FindRegressionIntersect(model, reference, treatment)
+    gradient <- .FindRegressionGradient(model, reference, treatment)
+    for (covariate_value in seq(from = start_x, to = end_x, by = (end_x - start_x) / divisions)) {
+      treatments <- c(treatments, treatment)
+      covariate_values <- c(covariate_values, covariate_value)
+      y_mins <- c(
+        y_mins,
+        intersect + (gradient * covariate_value) - (0.04 * ((covariate_value - 5) ^ 2) + 1)
```

Comment:
Can you explain what's happening here (unless it's something hard coded that will be changed later)

Reply:
This is just a placeholder until I properly gather the data from the model. That's likely to be a separate task.